### ISSUE TITLE
Error setting up entry Worx

### ISSUE THREAD

**Issue by tofrie** [MTrab/landroid_cloud#377]
### Describe the issue

Same bug with the new version.

### What version of Home Assistant Core has the issue?

2023.5.3

### What was the last working version of Home Assistant Core?

2023.5.3

### What version of the Landroid Cloud integration do you have installed

3.0.1

### What type of installation are you running?

Home Assistant OS

### Which make and model is the mower used for this integration?

Worx Landroid Vision M600

### Diagnostics information (NOT log entries!)

.

### Relevant log entries

```txt
Dieser Fehler wurde von einer benutzerdefinierten Integration verursacht

Logger: homeassistant.config_entries
Source: custom_components/landroid_cloud/__init__.py:206 
Integration: Landroid Cloud (documentation, issues) 
First occurred: 06:00:56 (2 occurrences) 
Last logged: 06:25:43

Error setting up entry Worx - xxx@xxx.de for landroid_cloud
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 387, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/config/custom_components/landroid_cloud/__init__.py", line 63, in async_setup_entry
    result = await _async_setup(hass, entry)
  File "/config/custom_components/landroid_cloud/__init__.py", line 206, in _async_setup
    await hass.async_add_executor_job(cloud.connect)
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/pyworxcloud/__init__.py", line 235, in connect
    self._fetch()
  File "/usr/local/lib/python3.10/site-packages/pyworxcloud/__init__.py", line 523, in _fetch
    self._decode_data(device)
  File "/usr/local/lib/python3.10/site-packages/pyworxcloud/__init__.py", line 339, in _decode_data
    device.zone.index = data["dat"]["lz"]
KeyError: 'lz'
```

**Comment by MTrab** [comment]
Please add diagnostics!

**Comment by MTrab** [comment]
And if possible, send credentials to landroid_cloud (at) trab.dk adding the text Issue 377 to the subject

**Comment by tofrie** [comment]
> Please add diagnostics!

There is nothing to download

**Comment by MTrab** [comment]
@frank-dusign not same error, but should be fixed in next release.

@tofrie this issue will be fixed with next release

**Comment by MTrab** [comment]
@tofrie - never mind ;) Just realised your mower is a Vision series.
These are not supported yet, as per #366

**Comment by MTrab** [comment]
> Looking forward to the fix, which has anounce already for 3.0.1 as it was reported in 3b6. Thanks

Yeah - there was one line I hade missed to put a check on. (Probably more, but they'll be fixed as soon as they are reported)
